### DISCUSSION TITLE
Duplicate styles in onRenderClient.css and onRenderHtml.css

### DISCUSSION THREAD

**Discussion by shishkin17** [vikejs/vike#1948] [Discussion]
There used to be just one file generated for styles - onRenderClient.css.

I updated vike and vite (used pretty old versions) and noticed that now it generates more css files including onRenderClient.css and **onRenderHtml**.css.
I use SSG model. So when I prerender the site I get 2 css files with almost the same content.

I am confused: is this doing vike or vite?
Can you please give a hint if I can control it somehow or make a workaround?

**Comment by brillout** [comment]
What's your `vike-react` version?

**Reply by shishkin17** [reply]
I use plain vike v0.4.199.

I found that vite is responsible for that. To disable multiple css files `cssCodeSplit: false` is needed. And it solves the problem.

But if someone wants to split css, then there will be an issue in case of prerender.

Sorry for long response time.

**Reply by brillout** [reply]
I did some digging, indeed `cssCodeSplit: false` leads to duplicated CSS. I've created and issue for it https://github.com/vikejs/vike/issues/1993 and added a warning.

**Reply by shishkin17** [reply]
Thank you. I'm glad I was useful in reporting that.

But cssCodeSplit: **true** leads to duplicated CSS.
So [this warning](https://github.com/vikejs/vike/commit/bc2d5a8dd1e75eed3d9daf5e86f7c21ad63afb1a) seems to work incorrectly.

**Reply by brillout** [reply]
> But cssCodeSplit: **true** leads to duplicated CSS.

Minimal reproduction welcome.

**Reply by shishkin17** [reply]
I mean:
`cssCodeSplit: false` = 1 css file - no problem
`cssCodeSplit: true` = multiple css files - duplication problem.

Do you want to say that while digging you found a problem when `cssCodeSplit: false`, not with `cssCodeSplit: true`?

**Reply by brillout** [reply]
> Do you want to say that while digging you found a problem when `cssCodeSplit: false`, not with `cssCodeSplit: true`?

Yes

**Reply by brillout** [reply]
Thus minimal reproduction welcome of having duplication problems with `cssCodeSplit: true`.

**Comment by richard-unterberg** [comment]
the build warning brought me here. I looked up the build files and the waterfall, cannot reproduce the duplication issue on fresh `vike + vike-solid` and `vite 7.*` setups. `cssCodeSplit: false / true` also works well for `vike-react` and tailwind and ssr/ssg activated. Also tested on multiple grown up projects with `vite 6.*`. Neither the content of the css file is duplicated, nor the imports.

I can confirm, there was a time I stumbled upon a similiar issue with a `vike-react` version, where there was a identical css file built with only one line of code difference. But that seems fixed in the more recent versions. [shishkin17](https://github.com/shishkin17) - have you tried newer versions?

@brillout

**Reply by richard-unterberg** [reply]
sorry, I had missed that [shishkin17](https://github.com/shishkin17) is using vike without react adapter.. Will test that tomorrow.

**Reply by brillout** [reply]
> the build warning brought me here

If you use the latest Vike version, you shouldn't be getting the warning `The CSS browser target should be the same for both client and server` anymore — AFAICT Vike is now clever enough to automatically fix it.

If you do get the warning, the best would be a minimal reproduction.

**Reply by richard-unterberg** [reply]
> If you use the latest Vike version, you shouldn't be getting the warning "The CSS browser target should be the same for both client and server anymore "

Ah sorry, I left you without context - I actually hit this warning `build.cssCodeSplit shouldn't be set to false (https://github.com/vikejs/vike/issues/1993)` which linked to the discussion here..

**Reply by brillout** [reply]
No problems. I've updated https://github.com/vikejs/vike/issues/1993 to be clearer — it now explains what the root cause is.

Let's wait for Vite to fully adopt Rolldown, let's then see if we can remove the warning.
